_posts/2016-11-14-JS-Glossary.md

```diff
@@ -0,0 +1,554 @@
+
+If you'd like to read more, there is a much deeper explanation in [uberVU/react-guide][react-guide]   
+
+### Context
+
+[The docs][context_docs] are pretty specific about context:
+
+> If you aren't an experienced React developer, don't use context. There is usually a better way to implement functionality just using props and state.
+
+Seems to be you should only be using this is really, really specific places. If you need it, you don't need this glossary.
+
+### JSX
+
+As we'll find out later, modern JavaScript is a collection of different ideas, and using [Babel](#babel) - you can add them at will into your projects. JSX is one such feature, it is a way of describing nested data using XML-like syntax. These are used inside React's render function to express a component's children and their [props](#props).
+
+Under the hood, JSX is quite simple, with code looking like this:
+
+```js
+const element = (
+  <h1 className="greeting">
+    Hello, world!
+  </h1>
+);
+```
+
+Turning into 
+
+```js
+const element = React.createElement(
+  'h1',
+  {className: 'greeting'},
+  'Hello, world!'
+);
+
+```
+
+Where `createElement` comes from the React [module](#module). You can find out more in [the React docs][react_jsx]
+
+# Libraries
+
+### GraphQL
+
+TLDR: An API format for requesting only the data you want, and getting back just that. 
+
+If you want the longer explanation, I wrote a [blog post on it](/blog/2016/06/19/graphql-for-mobile/).
+
+### Relay
+
+Relay is what makes working in our React Native app shine. It is a library that allows a component to describe the fragments of a networking request it would need to render. Relay would then look through your component hierarchy, take all the networking fragments, make a single GraphQL request and handle passing in the API fragments as [props](#props).
+
+This means you can throw away a significant amount of glue code.
+
+### Redux
+
+Redux is a state management pattern, it builds on top of React's "state is only passed down" concept, and creates a single way to handle triggering changes to your state. I'm afraid I don't have any experience with it, so I can't provide much context. I feel like [this post][what_redux] covers it well though.   
+
+# Tooling
+
+
+### Node
+
+Node is the JavaScript implementation from Google's Chrome (called v8) with an expanded API for doing useful systems tooling. It is a pretty recent creation, so it started off with an entirely synchronous API for any potentially blocking code.
+
+For web developers this was a big boon, you could share code between the browser and the server. The blocking API meant it was much easier to write faster servers, and there are lots of big companies putting a lot of time and money into improving the speed of JavaScript every day.
```

Comment:
`The blocking API meant it was much easier to write faster servers` I think you ment `non blocking API` in this paragraph?